tests/cases/fourslash/codeFixInferFromUsageJSXElement.ts

```diff
@@ -0,0 +1,40 @@
+/// <reference path='fourslash.ts' />
+
+// @noImplicitAny: true
+// @jsx: react
+// @module: es2015
+// @moduleResolution: node
+
+// @Filename: /node_modules/@types/react/index.d.ts
+////export = React;
+////export as namespace React;
+////declare namespace React {
+////    export class Component { render(): JSX.Element | null; }
+////}
+////declare global {
+////    namespace JSX {
+////        interface Element {}
+////    }
+////}
+
+
+// @filename: a.tsx
+//// import React from 'react';
+////
+//// export default function Component([|props |]) {
+////     if (props.isLoading) {
+////         return <div>...Loading < /div>;
+////     }
+////     else {
+////         return <AnotherComponent
+////             update={
+////             (rec) => {
+////                 return props.update(rec);
+////             }
+////         }
+////         />;
+////     }
+//// }
+
+
+verify.rangeAfterCodeFix("props: { isLoading: any; update: (...arg0: any) => void; }",/*includeWhiteSpace*/ undefined, /*errorCode*/ undefined, 0);
```

Comment:
Type is wrong. Should be `...arg0: any[]` that little bit not being implemented is probably why it was asserted.

Reply:
yeah, was looking into this

Reply:
In `maybeAddJsSyntheticRestParameter` we just to a switch to create an array type when building the last type if it's a rest type, probably just need to do a similar thing in `symbolToParameterDeclaration`.